src/json/json-document.ts

```diff
@@ -54,4 +54,8 @@ export class ParsedJsonDocument extends ParsedDocument<Json, Visitor> {
   sourceRangeForNode(_node: Json): SourceRange {
     throw new Error('Not Implemented.');
   }
+
+  stringify() {
+    return JSON.stringify(this.ast, null, 2);
```

Comment:
+1 to this or something similar. Writing a JSON parser is really easy if it comes to that.

Looks like what we'd do here is query the state of the input stream as we got events and either have a WeakMap alongside (if we only care about objects and arrays) or build up our own kind of annotated-node based tree?